numba/tests/npyufunc/test_dufunc.py

```diff
@@ -140,7 +141,369 @@ def impl():
+        add_at = self._generate_jit(np.add)
+        add_at(a, (slice(None), np.asarray([1, 2, 1])), b)
+        self.assertPreciseEqual(a, np.array(
+            [[0, 201, 102], [3, 404, 205], [6, 607, 308]]))
+
+        a = np.arange(27).reshape(3, 3, 3)
+        b = np.array([100, 200, 300])
+        add_at(a, (slice(None), slice(None), np.asarray([1, 2, 1])), b)
+        self.assertPreciseEqual(a, np.array(
+                                [[[0, 401, 202],
+                                  [3, 404, 205],
+                                  [6, 407, 208]],
+
+                                 [[9, 410, 211],
+                                  [12, 413, 214],
+                                  [15, 416, 217]],
+
+                                 [[18, 419, 220],
+                                  [21, 422, 223],
+                                  [24, 425, 226]]]))
+
+        a = np.arange(9).reshape(3, 3)
+        b = np.array([[100, 100, 100], [200, 200, 200], [300, 300, 300]])
+        add_at(a, (np.asarray([1, 2, 1]), slice(None)), b)
+        self.assertPreciseEqual(a, np.asarray(
+            [[0, 1, 2], [403, 404, 405], [206, 207, 208]]))
+
+        a = np.arange(27).reshape(3, 3, 3)
+        b = np.array([100, 200, 300])
+        add_at(a, (slice(None), np.asarray([1, 2, 1]), slice(None)), b)
+        self.assertPreciseEqual(a, np.asarray(
+                                [[[0,  1,  2],
+                                  [203, 404, 605],
+                                  [106, 207, 308]],
+
+                                 [[9,  10, 11],
+                                  [212, 413, 614],
+                                  [115, 216, 317]],
+
+                                 [[18, 19, 20],
+                                  [221, 422, 623],
+                                  [124, 225, 326]]]))
+
+        a = np.arange(9).reshape(3, 3)
+        b = np.array([100, 200, 300])
+        add_at(a, (0, np.asarray([1, 2, 1])), b)
+        self.assertPreciseEqual(a, np.asarray(
+            [[0, 401, 202], [3, 4, 5], [6, 7, 8]]))
+
+        a = np.arange(27).reshape(3, 3, 3)
+        b = np.array([100, 200, 300])
+        add_at(a, (np.asarray([1, 2, 1]), 0, slice(None)), b)
+        self.assertPreciseEqual(a, np.asarray(
+                                [[[0,  1,  2],
+                                  [3,  4,  5],
+                                  [6,  7,  8]],
+
+                                 [[209, 410, 611],
+                                  [12,  13, 14],
+                                  [15,  16, 17]],
+
+                                 [[118, 219, 320],
+                                  [21,  22, 23],
+                                  [24,  25, 26]]]))
+
+    @unittest.expectedFailure
+    def test_ufunc_at_multiD_unituple(self):
+        a = np.arange(27).reshape(3, 3, 3)
+        b = np.array([100, 200, 300])
+        add_at = self._generate_jit(np.add)
+        add_at(a, (slice(None), slice(None), slice(None)), b)
```

Comment:
This test is failing because the indexing variable is typed as `UniTuple(int64, 3)` and an attempt to `zip(indexing, [_, _, _])` will stop after the first iteration.